File: Session5/java.cpp
#include <iostream>
using namespace std;

class PaymentProcessor
{
public:

    // Method 1
    void processPayment(double amount)
    {
        cout << "Payment without coupon" << endl;
        cout << "Final Amount : " << amount << endl;
    }

    // Method 2 (Overloading)
    void processPayment(double amount, string coupon)
    {
        double finalAmount = amount - 100;

        cout << "Payment with coupon : " << coupon << endl;
        cout << "Final Amount : " << finalAmount << endl;
    }
};

int main()
{
    PaymentProcessor p;

    p.processPayment(1000);

    cout << endl;

    p.processPayment(1000, "SAVE100");

    return 0;
}

File: Session4/java.java
class PaymentProcessor
{
    // Method 1
    void processPayment(double amount)
    {
        System.out.println("Payment without coupon");
        System.out.println("Final Amount : " + amount);
    }

    // Method 2 (Overloaded)
    void processPayment(double amount, String coupon)
    {
        double finalAmount = amount - 100;

        System.out.println("Payment with coupon : " + coupon);
        System.out.println("Final Amount : " + finalAmount);
    }
}

public class Main
{
    public static void main(String args[])
    {
        PaymentProcessor p = new PaymentProcessor();

        p.processPayment(1000);

        p.processPayment(1000, "SAVE100");
    }
}
